src/crimson/osd/pg.h

```diff
@@ -952,6 +952,10 @@ class PG : public boost::intrusive_ref_counter<
       !peering_state.get_missing_loc().readable_with_acting(
 	oid, get_actingset(), v);
   }
+
+  // check if any head or clone of this object is missing
```

Comment:
This could have been also implemented by expanding PGLog interface to support `is_missing` on oid (not hoid).
Traversing seemed more straightforward (and didn't include changes in the common interface).

Only downside is possible large number of iterations when snap sequence is big. However, our missing set is a map so this lookup  (IMO) looks better than trying to serve a read on missing object (and only then failing).